linkatos/activities.py

```diff
@@ -3,29 +3,107 @@
 from . import firebase as fb
 from . import reaction as react
 from . import message
+from . import cache as cch
 
 
 def is_empty(events):
+    """
+    Checks if a batch of Slack events is `None` or has zero elements.
+    """
     return ((events is None) or (len(events) == 0))
 
 
-def is_url(element):
-    return element is not None
+def is_usable(element, bot_id):
+    """
+    Checks if there is a new element(url) that is usable, by verifying that is
+    not `None` and that it is not from the bot.
+    """
+    return element is not None and is_not_from_bot(bot_id, element['user'])
 
 
 def is_not_from_bot(bot_id, user_id):
-    return not bot_id == user_id
-
+    """
+    Compares two bot ids and verifies that they're different.
 
-def is_empty_list(xs):
-    return len(xs) == 0
+    is this really necessary??
+    """
+    return not bot_id == user_id
 
 
 def is_unfurled(event):
+    """
+    Checks if a Slack event comes from unfurling a url.
+
+    Currently assumes that any such message can be identified by the fact that
+    it contains a 'previous_message' field.
+    """
     return 'previous_message' in event
 
 
+def message_consumer(event, cache, slack_client, bot_id, fb_credentials,
+                     firebase):
+    """
+    Consumes an event of type 'message' that contains a user.
+    It looks inside the 'text' field for:
+        + a url
+        + @linkatos list
+        + @linkatos purge <i>
+       where <i> is the ith element of the cache
+    """
+    url = parser.parse_url_message(event)
+
+    if is_usable(url, bot_id):
+        return cch.add(url, cache, slack_client)
+
+    if message.to_bot(event['text'], bot_id):
+        list_request = parser.parse_list_request(event)
+        purge_request = parser.parse_purge_request(event)
+
+        if list_request is not None and list_request['type'] == 'list_request':
```

Comment:
Don't do it in this PR though, this will probably change after introducing some OOP.